src/Auth/Guards/LITokenGuard.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace LiveIntent\LaravelCommon\Auth\Guards;
+
+use Throwable;
+use Illuminate\Support\Arr;
+use Illuminate\Http\Request;
+use Lcobucci\JWT\Token\Plain;
+use Lcobucci\Clock\SystemClock;
+use Lcobucci\JWT\Configuration;
+use Lcobucci\JWT\Signer\Ecdsa\Sha256;
+use Lcobucci\JWT\Signer\Key\InMemory;
+use Illuminate\Contracts\Auth\UserProvider;
+use Illuminate\Contracts\Auth\Authenticatable;
+use Lcobucci\JWT\Validation\Constraint\SignedWith;
+use Lcobucci\JWT\Validation\Constraint\StrictValidAt;
+
+class LITokenGuard
+{
+    /**
+     * Configuration container for the JWT Builder and Parser.
+     */
+    private Configuration $configuration;
+
+    /**
+     * Create a new internal token guard instance.
+     * @throws Throwable
+     */
+    public function __construct(
+        private UserProvider $userProvider,
+        $plaintextPublicKey,
+    ) {
+        throw_if($plaintextPublicKey == null);
```

Comment:
1. Yeah, that makes sense. Wasn't sure if the better approach to null handling was to check within the function.

2. I was thinking that LITokenGuard shouldn't care how the public key is loaded. It would make this class a bit easier to write test for/with if that's needed somewhere without really needing to understand the inner workings of this class. However, I think to facilitate the ease of use and it working "out of the box", it makes it a bit easier to not have it as a constructor value. I'm fine with either approach if you prefer one way over another for consistency. I'll remove it from the constructor since we don't have any tests for this anyway right now.

Reply:
There are nice ways to test this even without the key being passed through the constructor, since it's still passed through the config, I can show you.